cmake/cmakelibs.cmake

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2014-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+
+# This function takes the global properties saved by add_osquery_extension_ex and 
+# generates a single extension executable containing all the user code
+function(generate_osquery_extension_group)
+  get_property(extension_source_files GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_SOURCES)
+  if("${extension_source_files}" STREQUAL "")
+    return()
+  endif()
+
+  # Allow the user to customize the extension name and version using environment variables
+  if(DEFINED ENV{OSQUERY_EXTENSION_GROUP_NAME})
```

Comment:
I would convert this into a cache variable, leaving though a "deprecated" warning if set, then falling back to the cache variable.

Reply:
OSQUERY_EXTENSION_GROUP_{NAME/VERSION} is defined as a cache variable and can be set by the extension instead of using the env variable.